src/Controls/GlobeControls.js

```diff
@@ -521,60 +542,145 @@ class GlobeControls extends THREE.EventDispatcher {
         }
     }
 
-    onMouseMove(event) {
+    initMovement() {
+        CameraUtils.stop(this.view, this.camera);
+        this.player.stop();
+        this.onEndingMove();
+
+        this.updateTarget();
+        previous = CameraUtils.getTransformCameraLookingAtTarget(this.view, this.camera, pickedPosition);
+    }
+
+    initRotation(event) {
+        this.initMovement();
+
+        this.state = this.states.ORBIT;
+        rotateStart.copy(event.viewCoords);
+    }
```

Comment:
Could you merge all inits in one method?

Reply:
Could you move `this.state` to `StateControl`?

Reply:
> Could you merge all inits in one method?

As mentioned above, I added a 'state-changed' event to manage all initialization.



> Could you move `this.state` to `StateControl`?

For viewers information : moving `this.state` to `StateControl` poses a few issues with damping, which are curenty being worked around.